models/accm.py

```diff
@@ -30,14 +31,24 @@ def forward(self, x):
 
 
 class AccmBlock(nn.Module):
-    def __init__(self, input_layer, h_channels, number_of_rsacm):
+    def __init__(self, input_layer, h_channels, number_of_rsacm, args, rsacm_cost):
         super().__init__()
         self.h_channels = h_channels
         self.number_of_rsacm = number_of_rsacm
+        self.args = args
         self.rsacm_list = nn.ModuleList(
             [(RsacmBlock(input_layer, self.h_channels)) for i in range(0, self.number_of_rsacm)])
+        self.rsacm_cost = rsacm_cost
+
+    def initialize_gating_network(self, activation, args):
+        self.gating_network =  Gating_Network(activation, args)
+        self.optimizer_for_gating_network = torch.optim.Adam(self.gating_network.parameters(), args.lr, weight_decay=0)

```

Comment:
Looks like this is redundant and unused now?

Reply:
So 
`self.gating_network` is used but `self.optimizer_for_gating_network` is not used. I will remove `self.optimizer_for_gating_network` when doing clean-up.